gwpy/timeseries/timeseries.py

```diff
@@ -986,7 +990,9 @@ def zpk(self, zeros, poles, gain, analog=True, **kwargs):
 
         >>> data2 = data.zpk([100]*5, [1]*5, 1e-10)
         """
-        return self.filter(zeros, poles, gain, analog=analog, **kwargs)
+
+        f = self.filter(zeros, poles, gain, analog=analog, unit=unit, **kwargs)
+        return f
```

Comment:
Is this change necessary?

Reply:
I think it was a line length workaround.

Reply:
Can this be reverted? Assigning to a variable is unnecessary here. If the line is too long, just wrap:

```suggestion
        return self.filter(
            zeros,
            poles,
            gain,
            analog=analog,
            unit=unit,
            **kwargs,
        )
```